samples/features/security/ledger/azure-sql-database/setup/setup.ps1

```diff
@@ -0,0 +1,118 @@
+$sqlAdminPasswordSecureString = Read-Host -Prompt "Enter the password of the Azure SQL database server administrator for SQL authentication" -AsSecureString
+
+$sqlAdminPassword = (New-Object PSCredential "user",$sqlAdminPasswordSecureString).GetNetworkCredential().Password
+$clientIP = (Invoke-WebRequest ifconfig.me/ip).Content.Trim()
+$bicepFile = "azuredeploy.bicep"
+$projectName = $projectName.ToLower()
+$location = $location.ToLower()
+
+######################################################################
+# Sign in to Azure
+######################################################################
+
+Connect-AzAccount
+$context = Set-AzContext -Subscription $subscriptionId
+$userName = $context.Account.Id
+$userObjectId = $(Get-AzADUser -UserPrincipalName $userName).Id
+
+######################################################################
+# Create a resource group
+######################################################################
+$resourceGroupName = "${projectName}"
+New-AzResourceGroup -Name $resourceGroupName -Location $location
+
+######################################################################
+# Deploy the resources for the demo environment
+######################################################################
+
+$deployment=New-AzResourceGroupDeployment `
+  -ResourceGroupName $resourceGroupName `
+  -TemplateFile $bicepFile `
+  -projectName $projectName `
+  -userObjectId $userObjectId `
+  -userName $userName `
+  -sqlAdminUserName $sqlAdminUserName `
+  -sqlAdminPassword $sqlAdminPassword `
+  -clientIP $clientIP
+
+$fullServerName = $deployment.Outputs.servername.Value
+$databaseName = $deployment.Outputs.databaseName.Value
+$appName = $deployment.Outputs.appname.Value
+
+######################################################################
+# Create the database schema and populate it with data
+######################################################################
+
+# Create the databse schema
+$queryFile = "CreateDatabaseSchema.sql"
+$query = Get-Content -path $queryFile -Raw
+$accessToken = (Get-AzAccessToken -ResourceUrl https://database.windows.net).Token
+Invoke-Sqlcmd -ServerInstance "tcp:$fullServerName" -Database $databaseName -AccessToken $accessToken -QueryTimeout 30 -Query $query
+
+# Populate the database
+$queryFile = "PopulateDatabase.sql"
+$query = Get-Content -path $queryFile -Raw
+$accessToken = (Get-AzAccessToken -ResourceUrl https://database.windows.net).Token
+Invoke-Sqlcmd -ServerInstance "tcp:$fullServerName" -Database $databaseName -AccessToken $accessToken -QueryTimeout 30 -Query $query
+
+######################################################################
+# Grant the web application access to the database
+######################################################################
+
+# Create a shadow principal, representing the application, in the database
+$query = "CREATE USER [$appName] FROM EXTERNAL PROVIDER;"
```

Comment:
1. You could construct the statements in a batch and run one Invoke-Sqlcmd
2. Change sp_addrolemember to ALTER ROLE. Ex: ALTER ROLE db_datareader ADD MEMBER [$appName]
3. Now, you can treat the names as SQL identifier and escape only ] to ]] inside the identifier. This will handle case where user input [] in the app name. You can use the string interpolation operator --replace like $appName -replace ']', ']]'

Reply:
1. Indeed, but we wanted to make it clear for the user that deploys the demo what happened in the script. first, deployment of the DB schema and secondly populate the DB.
2. Done
3. When I change [$appName] into [[$appName]] I get a syntax error. So, I left the original code for now